src/api/syscall.c

```diff
@@ -396,20 +396,26 @@ static exception_t handleInvocation(bool_t isCall, bool_t isBlocking)
         return EXCEPTION_NONE;
     }
 
-    buffer = lookupIPCBuffer(false, thread);
-
-    status = lookupExtraCaps(thread, buffer, info);
-
-    if (unlikely(status != EXCEPTION_NONE)) {
-        userError("Lookup of extra caps failed.");
-        if (isBlocking) {
-            handleFault(thread);
+    buffer = NULL;
+    length = seL4_MessageInfo_get_length(info);
```

Comment:
No such thing as a zero-length buffer, right? If length is zero, the pointer is NULL.
Otherwise, `src/api/syscall.c:419` semantics would change.